### PR TITLE
Adds transform path for region coder

### PR DESCRIPTION
**Pull request by opleban**
Adds separate regionCodeByTransform with the ability to return text values in addition to numbers.

The new function takes in an array of geospatial points and return an array of labels corresponding to the features in which those points lie. E.g.: an array of county names.

Currently region coder will return an numeric id of a feature. For example if you send it a point it will return the numeric id of the polygon feature (say a county id) it falls in. This change will enable region coder to also return a string label for the polygon feature, such as the county name.

### CODE REVIEW

File: src/main/scala/com/socrata/regioncoder/RegionCoder.scala [hunk 1]
```diff
@@ -41,11 +42,38 @@ trait RegionCoder {
...
+    )
+    // Map unique partitions to SpatialIndices, fetching them in parallel using Futures
+    // Now we have a Seq[Future[Envelope -> SpatialIndex]]
+    val indexFutures = partitions.toSet.map { partEnvelope: Envelope =>
+      spatialCache.getFromSoda(sodaFountain, resourceName, idFeatureColumn, Some(partEnvelope))
+        .map(partEnvelope -> _)
+    }
+    // Turn sequence of futures into one Future[Map[Envelope -> SpatialIndex]]
+    // which will be done when all the indices/partitions have been fetched
+    Future.sequence(indexFutures).map(_.toMap).map { envToIndex =>
```
Comment (rozap) [inline src/main/scala/com/socrata/regioncoder/RegionCoder.scala:65]: are we bounding this parallelism at all? if not, we should

**Comment by znep** [comment]
This doesn't seem to be building in jenkins, is it missing some of the code?

**Comment by catstavi** [comment]
If the main goal of the new path is to be compatible with id columns that are strings instead of numbers, I'd update the name of what you're building to focus on that!

File: src/main/scala/com/socrata/regioncoder/RegionCoder.scala (hunk 1 above)
Comment (opleban) [reply src/main/scala/com/socrata/regioncoder/RegionCoder.scala:65]: Do you have an example of somewhere else we do this, I could crib off?

**Comment by opleban** [comment]
> This doesn't seem to be building in jenkins, is it missing some of the code?

🤔 It builds and runs locally for me. Let me take another look.

**Comment by opleban** [comment]
> If the main goal of the new path is to be compatible with id columns that are strings instead of numbers, I'd update the name of what you're building to focus on that!

@catstavi that's fair. Originally, I had it in mind to have regionCodeByTransform handle all DSMAPI data transforms requests.

File: src/main/scala/com/socrata/regioncoder/RegionCoder.scala (hunk 1 above)
Comment (opleban) [reply src/main/scala/com/socrata/regioncoder/RegionCoder.scala:75]: Assuming the PR that @rjmac merged in yesterday addressed this, I went ahead and made a similar change to LabelCache.

**Pull request closed without merging**